lib/tbot/workloadidentity/workloadattest/unix_linux.go

```diff
@@ -0,0 +1,83 @@
+//go:build linux
+
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package workloadattest
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/gravitational/trace"
+	"github.com/shirou/gopsutil/v4/process"
+)
+
+var unixOS UnixOS = linux{}
+
+type linux struct{}
+
+func (l linux) ExePath(ctx context.Context, proc *process.Process) (string, error) {
+	// If the workload process is in a different mount namespace (e.g. running
+	// in a container) the path we see will likely be wrong.
+	sameNS, err := l.sameMountNS(proc.Pid)
+	if err != nil {
+		return "", err
+	}
+	if !sameNS {
+		return "", trace.NotFound("unable to find executables in a different mount namespace")
+	}
+	return proc.ExeWithContext(ctx)
+}
+
+func (l linux) OpenExe(ctx context.Context, proc *process.Process) (io.ReadCloser, error) {
+	// On Linux, `/proc/<pid>/exe` is a symlink to the *inode* of the process'
+	// executable rather than a simple path, this means it won't change even if
+	// you replace the file on disk.
+	//
+	// In other words, during a rolling deployment the binary's hash won't change
+	// until the process has actually been restarted - which is desirable.
+	//
+	// With one important caveat: network filesystems typically do not guarantee
+	// inode stability, so if the process' binary is on a network mount, it's
+	// possible the hash won't match the binary the process is actually running.
+	return os.Open(l.procPath(strconv.Itoa(int(proc.Pid)), "exe"))
+}
+
+func (l linux) sameMountNS(pid int32) (bool, error) {
+	tbotNS, err := os.Readlink(l.procPath("self", "ns", "mnt"))
+	if err != nil {
+		return false, trace.Wrap(err, "reading tbot mount namespace")
+	}
+	procNS, err := os.Readlink(l.procPath(strconv.Itoa(int(pid)), "ns", "mnt"))
+	if err != nil {
+		return false, trace.Wrap(err, "reading the workload process mount namespace")
+	}
+	return tbotNS == procNS, nil
+}
```

Comment:
This only works if `tbot` is running as the same user as the workload or as root. I wonder if we should make the check best-effort and if we can't open `ns/mnt` log a warning and assume we're in the same namespace?

Reply:
Sounds good to me I think.

Reply:
I've decided to remove this check entirely because:

1. In a "good" configuration (i.e. `tbot` running as a non-root user, separate to the workload user) it will _always_ fail
2. If we log a warning, it could encourage users to change their configuration

Instead, let's make sure the documentation reflects that `binary_path` is unreliable in containerized workloads.